tests/test_others_dialect_specific.py

```diff
@@ -137,3 +137,40 @@ def test_lateral_view_outer(dialect: str):
 @pytest.mark.parametrize("dialect", ["databricks", "sparksql"])
 def test_show_create_table(dialect: str):
     assert_table_lineage_equal("show create table tab1", None, None, dialect)
+
+
+@pytest.mark.parametrize("dialect", ["bigquery"])
+def test_merge_using_subquery(dialect: str):
+    sql = """MERGE target USING (select k, max(v) as v from src group by k) AS b ON target.k = b.k
+WHEN MATCHED THEN UPDATE SET target.v = b.v
+WHEN NOT MATCHED THEN INSERT (k, v) VALUES (b.k, b.v)"""
+    assert_table_lineage_equal(sql, {"src"}, {"target"}, dialect=dialect)
+
+
+@pytest.mark.parametrize("dialect", ["bigquery"])
+def test_merge_using_cte_subquery(dialect: str):
+    sql = """MERGE `project_id.dataset_id.target_table` t
+    USING (
+    WITH base AS (
+        SELECT
+            date, channel, cnt, user_count, total_user_count,
+        FROM `project_id.dataset_id.origin_table`
+    )
+    SELECT
+        date, channel, cnt, total_user_count, SAFE_DIVIDE(cnt, user_count) AS rate
+    FROM base
+    ) s
+    ON t.date = s.date and t.channel = s.channel
+    WHEN NOT MATCHED THEN
+    INSERT ROW
+    WHEN MATCHED THEN
+    UPDATE SET t.total_user_cnt = s.total_user_cnt,
+    t.cnt = s.cnt,
+    t.rate = s.rat"""
+    assert_table_lineage_equal(
+        sql,
+        {"project_id.dataset_id.origin_table"},
+        {"project_id.dataset_id.target_table"},
+        dialect=dialect,
+        test_sqlparse=False,
+    )
```

Comment:
What's the specialty about this test case? It seems to me:

- MERGE without INTO is covered by test_merge_using_subquery
- MERGE using CTE is covered by test_merge_using_cte_subquery
- qualified identifier `project_id.dataset_id.target_table` is covered by test_select_with_table_name_in_backtick in test_select_dialect_specific.py

If no, then we can have this test case removed. If yes, let's narrow it down to the feature we want test. I'd like the test case to be as atomic as possible.

Reply:
Re-purposing this test case to be testing INSERT ROW.